WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -4,26 +4,42 @@ import Gridicons
 
 final class AuthenticatedImageDownload: AsyncOperation {
     let url: URL
-    let blog: Blog
+    let blogObjectID: NSManagedObjectID
     private let callbackQueue: DispatchQueue
     private let onSuccess: (UIImage) -> ()
     private let onFailure: (Error) -> ()
 
-    init(url: URL, blog: Blog, callbackQueue: DispatchQueue, onSuccess: @escaping (UIImage) -> (), onFailure: @escaping (Error) -> ()) {
+    init(url: URL, blogObjectID: NSManagedObjectID, callbackQueue: DispatchQueue, onSuccess: @escaping (UIImage) -> (), onFailure: @escaping (Error) -> ()) {
         self.url = url
-        self.blog = blog
+        self.blogObjectID = blogObjectID
         self.callbackQueue = callbackQueue
         self.onSuccess = onSuccess
         self.onFailure = onFailure
     }
 
     override func main() {
-        let mediaRequestAuthenticator = MediaRequestAuthenticator()
-        let host = MediaHost(with: blog) { error in
-            // We'll log the error, so we know it's there, but we won't halt execution.
-            WordPressAppDelegate.crashLogging?.logError(error)
+        let result = ContextManager.shared.performQuery { context in
+            Result {
+                let blog = try context.existingObject(with: self.blogObjectID) as! Blog
+                return MediaHost(with: blog) { error in
+                    // We'll log the error, so we know it's there, but we won't halt execution.
+                    WordPressAppDelegate.crashLogging?.logError(error)
```

Comment:
Out of scope and already existing code: This singleton way of logging... 😱 I appreciated it's convenient, but it'll make for extra work to isolate the components. Not to mention introducing implicit dependencies.